core/src/test/scala/org/apache/spark/sql/delta/DeltaDDLSuite.scala

```diff
@@ -476,4 +473,47 @@ abstract class DeltaDDLTestBase extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("table creation with ambiguous paths only allowed with legacy flag") {
+    // ambiguous paths not allowed
+    withTempDir { foo =>
+      withTempDir { bar =>
+          val fooPath = foo.getCanonicalPath()
+          val barPath = bar.getCanonicalPath()
+          val e = intercept[AnalysisException] {
+            sql(s"CREATE TABLE delta.`$fooPath`(id LONG) USING delta LOCATION '$barPath'")
+          }
+          assert(e.message.contains("DELTA_LEGACY_ALLOW_AMBIGUOUS_PATHS"))
+      }
+    }
+
+    // allowed with legacy flag
+    withTempDir { foo =>
+      withTempDir { bar =>
+        val fooPath = foo.getCanonicalPath()
+        val barPath = bar.getCanonicalPath()
+        withSQLConf(DeltaSQLConf.DELTA_LEGACY_ALLOW_AMBIGUOUS_PATHS.key -> "true") {
+          sql(s"CREATE TABLE delta.`$fooPath`(id LONG) USING delta LOCATION '$barPath'")
+          assert(io.delta.tables.DeltaTable.isDeltaTable(fooPath))
+          assert(!io.delta.tables.DeltaTable.isDeltaTable(barPath))
+        }
+      }
+    }
+
+    // allowed if paths are the same
+    withTempDir { foo =>
+      val fooPath = foo.getCanonicalPath()
+      withSQLConf(DeltaSQLConf.DELTA_LEGACY_ALLOW_AMBIGUOUS_PATHS.key -> "true") {
+        sql(s"CREATE TABLE delta.`$fooPath`(id LONG) USING delta LOCATION '$fooPath'")
+        assert(io.delta.tables.DeltaTable.isDeltaTable(fooPath))
+      }
+    }
+
+    // make sure default is false
+    assert(!spark.conf.get(DeltaSQLConf.DELTA_LEGACY_ALLOW_AMBIGUOUS_PATHS))
```

Comment:
we don't need this. The test that verifies the behavior without setting the flag has tested the default value.